plugins/inputs/dpdk/dpdk.go

```diff
@@ -96,136 +82,236 @@ func (dpdk *dpdk) Init() error {
+	}
+
+	if dpdk.DeviceTypes == nil {
+		dpdk.DeviceTypes = []string{"ethdev"}
+		dpdk.Log.Debugf("Using default %q value for device_types", dpdk.DeviceTypes)
+	}
+
+	if dpdk.MetadataFields == nil {
+		dpdk.MetadataFields = []string{dpdkMetadataFieldPidName}
+		dpdk.Log.Debugf("Using default %q value for metadata_fields", dpdk.MetadataFields)
+	}
+
+	if dpdk.PluginOptions == nil {
+		dpdk.PluginOptions = []string{dpdkPluginOptionInMemory}
+		dpdk.Log.Debugf("Using default %q value for plugin_options", dpdk.PluginOptions)
+	}
+
+	if len(dpdk.UnreachableSocketBehavior) == 0 {
+		dpdk.UnreachableSocketBehavior = unreachableSocketBehaviorError
+		dpdk.Log.Debugf("Using default %q value for unreachable_socket_behavior", unreachableSocketBehaviorError)
+	}
+
+	dpdk.rawdevCommands = []string{"/rawdev/xstats"}
+	dpdk.ethdevCommands = []string{"/ethdev/stats", "/ethdev/xstats", "/ethdev/info", ethdevLinkStatusCommand}
 }
 
 // Checks that user-supplied commands are unique and match DPDK commands format
-func (dpdk *dpdk) validateCommands() error {
+func (dpdk *dpdk) validateAdditionalCommands() error {
 	dpdk.AdditionalCommands = uniqueValues(dpdk.AdditionalCommands)
 
-	for _, commandWithParams := range dpdk.AdditionalCommands {
-		if len(commandWithParams) == 0 {
+	for _, cmd := range dpdk.AdditionalCommands {
+		if len(cmd) == 0 {
 			return fmt.Errorf("got empty command")
 		}
 
-		if commandWithParams[0] != '/' {
-			return fmt.Errorf("%q command should start with '/'", commandWithParams)
+		if cmd[0] != '/' {
+			return fmt.Errorf("%q command should start with '/'", cmd)
 		}
 
-		if commandWithoutParams := stripParams(commandWithParams); len(commandWithoutParams) >= maxCommandLength {
+		if commandWithoutParams := stripParams(cmd); len(commandWithoutParams) >= maxCommandLength {
 			return fmt.Errorf("%q command is too long. It shall be less than %v characters", commandWithoutParams, maxCommandLength)
 		}
 
-		if len(commandWithParams) >= maxCommandLengthWithParams {
-			return fmt.Errorf("command with parameters %q shall be less than %v characters", commandWithParams, maxCommandLengthWithParams)
+		if len(cmd) >= maxCommandLengthWithParams {
+			return fmt.Errorf("command with parameters %q shall be less than %v characters", cmd, maxCommandLengthWithParams)
 		}
 	}
 
 	return nil
 }
 
-// Gathers all unique commands and processes each command sequentially
-// Parallel processing could be achieved by running several instances of this plugin with different settings
-func (dpdk *dpdk) Gather(acc telegraf.Accumulator) error {
-	// This needs to be done during every `Gather(...)`, because DPDK can be restarted between consecutive
-	// `Gather(...)` cycles which can cause that it will be exposing different set of metrics.
-	commands := dpdk.gatherCommands(acc)
+// Establishes connections do DPDK telemetry sockets
+func (dpdk *dpdk) maintainConnections() error {
+	socketsToConnect, socketsToDisconnect := dpdk.identifySockets()
+
```

Comment:
I'm still thinking that you should merge this function in here as it will save you some more iterations over the sockets...

Reply:
Addressed in: https://github.com/influxdata/telegraf/pull/14308/commits/594fecb38d9f68e33ff03a50172ee132bc5811f6